openapi/openapi-shard-snapshots.ytt.yaml

```diff
@@ -0,0 +1,214 @@
+#@ load("openapi.lib.yml", "response", "response_with_accepted", "reference", "type", "array")
+
+paths:
+  /collections/{collection_name}/shards/{shard_id}/snapshots/upload:
+    post:
+      tags:
+        - snapshots
+        - collections
+      summary: Recover shard from an uploaded snapshot
+      description: Recover shard of a local collection from an uploaded snapshot. This will overwrite any data, stored on this node, for the collection shard.
+      operationId: recover_shard_from_uploaded_snapshot
+      parameters:
+        - name: collection_name
+          in: path
+          description: Name of the collection
+          required: true
+          schema:
+            type: string
+        - name: shard_id
+          in: path
+          description: Id of the shard to recover
+          required: true
+          schema:
+              type: integer
+        - name: wait
+          in: query
+          description: "If true, wait for changes to actually happen. If false - let changes happen in background. Default is true."
+          required: false
+          schema:
+            type: boolean
+        - name: priority
+          in: query
+          description: "Defines source of truth for snapshot recovery"
+          required: false
+          schema:
+            $ref: "#/components/schemas/SnapshotPriority"
+      requestBody:
+        description: Snapshot to recover from
+        content:
+          multipart/form-data:
+            schema:
+              type: object
+              properties:
+                snapshot:
+                  type: string
+                  format: binary
+      responses: #@ response_with_accepted(type("boolean"))
+  /collections/{collection_name}/shards/{shard_id}/snapshots/recover:
+    put:
+      tags:
+        - snapshots
+        - collections
+      summary: Recover from a snapshot
+      description: Recover shard of a local collection data from a snapshot. This will overwrite any data, stored in this shard, for the collection.
+      operationId: recover_shard_from_snapshot
+      parameters:
+        - name: collection_name
+          in: path
+          description: Name of the collection
+          required: true
+          schema:
+            type: string
+        - name: shard_id
+          in: path
+          description: Id of the shard to recover
+          required: true
+          schema:
+            type: integer
+        - name: wait
+          in: query
+          description: "If true, wait for changes to actually happen. If false - let changes happen in background. Default is true."
+          required: false
+          schema:
+            type: boolean
+      requestBody:
+        description: Snapshot to recover from
+        content:
+          application/json:
+            schema:
+              $ref: "#/components/schemas/SnapshotRecover"
```

Comment:
`recover_shard_snapshot` uses `ShardSnapshotRecover`, which is a bit different from `SnapshotRecover`, because we decided not to use `file://` URLs for shard snapshot API.

See:
- https://github.com/qdrant/qdrant/blob/dev/src/actix/api/snapshot_api.rs#L323
- https://github.com/qdrant/qdrant/blob/dev/src/actix/api/snapshot_api.rs#L549-L562
- https://github.com/qdrant/qdrant/blob/dev/src/actix/api/snapshot_api.rs#L334-L341